tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,256 @@
+import { FilterSearch } from '../../src/components/FilterSearch';
+import { render, screen } from '@testing-library/react';
+import * as useSynchronizedRequestFunctions from '../../src/hooks/useSynchronizedRequest';
+import userEvent from '@testing-library/user-event';
+import * as searchOperations from '../../src/utils/search-operations';
+import { mockAnswersActions } from '../__utils__/mocks';
+import { mockedFilterSearchResponse, mockedFilterSearchResponseNoResults, mockedFilterSearchResponseNoLabels} from '../../tests/__fixtures__/data/filtersearch';
+
+jest.mock('@yext/answers-headless-react');
+
+const mockedExecuteFilterSearch = jest.fn();
+const setFilterOption = jest.fn();
+const setOffset = jest.fn();
+const searchFieldsProp = {
+  fieldApiName: 'name',
+  entityType: 'ce-person'
+};
+
+describe('tests using data with section labels', () => {
+  beforeEach(() => {
+    mockAnswersActions({
+      setFilterOption,
+      setOffset
+    });
+
+    jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+    jest.spyOn(useSynchronizedRequestFunctions, 'useSynchronizedRequest').mockImplementation(() => {
+      return [mockedFilterSearchResponse, mockedExecuteFilterSearch, null];
+    });
```

Comment:
Instead of mocking `useSynchronizedRequestFunctions`, if possible, I think it may be better if we just directly mock executeFilterSearch in `mockAnswersActions`?

Reply:
I originally played around with that idea, but since I needed to also mock a value for filterSearchResponse which is reliant on useSynchronizedSearch, I figured that I would have to mock useSynchronizedSearch anyways. So I just mocked useSynchronizedSearch instead of both useSynchronizedSearch and executeFilterSearch.

Is the direct mocking through AnswersActions preferable?  I guess by mocking useSynchronizedSearch, we miss out on testing the logic inside the function call.

Reply:
Right, I think it's best if we can keep the logic of the hook used in FilterSearch as much as possible and only mock what is necessary. You can mock the return value filterSearchResponse for executeFilterSearch like so:
```
    mockAnswersActions({
      setFilterOption,
      setOffset,
      executeFilterSearch: jest.fn().mockResolvedValue(unsectionedFilterSearchResponse)
    });
```
And to spy on executeFilterSearch in your it() test, you can do something like:
```
    const actions = spyOnActions();
    expect(actions.executeFilterSearch).toHaveBeenCalledWith(blah);
```
You may encounter some warnings about wrapping stuff in act since there's some async request/response happening. So you may need `waitFor` in some of your assertions

Reply:
Got it-- changed to use mockAnswersActions and spyOnActions